praw/models/reddit/subreddit.py

```diff
@@ -670,9 +687,22 @@ def _upload_media(self, media_path, expected_mime_prefix=None, gallery=False):
         if not response.ok:
             self._parse_xml_response(response)
         response.raise_for_status()
-        if gallery:
+        if upload_type == "link":
+            return f"{upload_url}/{upload_data['key']}"
+        elif upload_type in ["gallery", "selfpost"]:
```

Comment:
I think this should just be `else`.
```suggestion
        else:
```